libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp

```diff
@@ -498,14 +505,27 @@ void AP_InertialSensor_Backend::update_gyro(uint8_t instance)
         _last_gyro_filter_hz[instance] = _gyro_filter_cutoff();
     }
     // possily update the notch filter parameters
-    if (_last_notch_center_freq_hz[instance] != _gyro_notch_center_freq_hz() ||
-        _last_notch_bandwidth_hz[instance] != _gyro_notch_bandwidth_hz() ||
+    if (!is_equal(_last_notch_center_freq_hz[instance], _gyro_notch_center_freq_hz()) ||
+        !is_equal(_last_notch_bandwidth_hz[instance], _gyro_notch_bandwidth_hz()) ||
         !is_equal(_last_notch_attenuation_dB[instance], _gyro_notch_attenuation_dB())) {
         _imu._gyro_notch_filter[instance].init(_gyro_raw_sample_rate(instance), _gyro_notch_center_freq_hz(), _gyro_notch_bandwidth_hz(), _gyro_notch_attenuation_dB());
         _last_notch_center_freq_hz[instance] = _gyro_notch_center_freq_hz();
         _last_notch_bandwidth_hz[instance] = _gyro_notch_bandwidth_hz();
         _last_notch_attenuation_dB[instance] = _gyro_notch_attenuation_dB();
     }
+    // possily update the harmonic notch filter parameters
+    if (!is_equal(_last_harmonic_notch_center_freq_hz[instance], _gyro_harmonic_notch_center_freq_hz())) {
+        _imu._gyro_harmonic_notch_filter[instance].update(_gyro_harmonic_notch_center_freq_hz());
+        _last_harmonic_notch_center_freq_hz[instance] = _gyro_harmonic_notch_center_freq_hz();
+    } 
+    // possily update the harmonic notch filter parameters
+    if (!is_equal(_last_harmonic_notch_bandwidth_hz[instance], _gyro_harmonic_notch_bandwidth_hz()) ||
```

Comment:
This *if* should be before the one above, with the one above being an else to this one. Reason is that if entering this *if* then center frequency will already be reset (and if not, then check if it's needed to update the center frequency).

Reply:
Ok, I just copied the previous code, will fix :)